flow.c

```diff
@@ -588,12 +588,7 @@ static void FlowInsert(u32 CPUID, FlowIndex_t* FlowIndex, FlowRecord_t* FlowPkt,
 
 	F->TotalFCS		+= FlowPkt->TotalFCS;
 
-	if (F->PktInfoB == NULL)
-	{
-		// allocate space for 1024 pkts histograms to avoid frequent malloc
-		F->PktInfoB = PktInfoBulk_Alloc(1024);
-	}
-	PktInfo_Insert(F->PktInfoB, Length, TS - F->FirstTS);
+	PktInfo_Insert(&F->PktInfoB, Length, TS - F->FirstTS);
```

Comment:
This needs to be conditional on g_Output_Histogram  otherwise the performance will drop significantly running in normal operation mode.

Reply:
sending TS - F->LastTS is better here, or tracking the last TS in PacketInfoBulk. Using delta from FirstTS means thres a max of 4sec (32bit) which wont be long enough I think.

Reply:
> sending TS - F->LastTS is better here, or tracking the last TS in PacketInfoBulk. Using delta from FirstTS means thres a max of 4sec (32bit) which wont be long enough I think.

hmm...u32 will not be sufficient...But the issue remain even with "LastTS".
How about if I store TS (u64) for each packet ? It will take ~ 3.8 MB extra space for 1M packets.

Reply:
If you store the delta TS - TSLast then it maximum 4sec between packets instead of since the 1st packet. Dont think its worth the storage cost to go to u64, maybe max it at 4sec / 32bit?

Reply:
But what if 2 packets in the same flow has more than 4 seconds gap ?

I understood by using LastTS, we will store the TS difference between two subsequent packets. Which will serve our need in most of the cases but some corner cases where two subsequent packet of the flow has more than ~4 seconds gap then it will again have the 32bit max limitation.

Reply:
in such case the time is > 32b, set the dTime to be (1<<32)-1

not perfect but its worth storage cost saving I think